CI failures - Taylor diagram legend location

It looks like we're getting some CI failures: https://github.com/NCAR/geocat-viz/actions/runs/13710707456

They seem to have started with Matplotlib v3.10.1, but I haven't gotten much further than that in tracking.

Here's a snippet from the relevant log:
<details>
```
=================================== FAILURES ===================================
_______________________________ test_add_legend ________________________________
Error: Image files did not match.
  RMS Value: 10.8119[21](https://github.com/NCAR/geocat-viz/actions/runs/13710707456/job/38346433861#step:6:22)630003306
  Expected:  
    /var/folders/0j/bwqcs4y508s2n4ck4dhf3rpc0000gn/T/tmptwuhdz64/tests.test_taylor.test_add_legend/baseline.png
  Actual:    
    /var/folders/0j/bwqcs4y508s2n4ck4dhf3rpc0000gn/T/tmptwuhdz64/tests.test_taylor.test_add_legend/result.png
  Difference:
    /var/folders/0j/bwqcs4y508s2n4ck4dhf3rpc0000gn/T/tmptwuhdz64/tests.test_taylor.test_add_legend/result-failed-diff.png
  Tolerance: 
    2
=============================== warnings summary ===============================
tests/test_taylor.py::test_add_model_set
tests/test_taylor.py::test_add_legend
tests/test_taylor.py::test_add_bias_legend
tests/test_taylor.py::test_add_model_name
tests/test_taylor.py::test_add_corr_grid
tests/test_taylor.py::test_add_contours
  /Users/runner/miniconda3/envs/geocat_viz_build/lib/python3.12/site-packages/geocat/viz/taylor.py:103: MatplotlibDeprecationWarning: Passing `apply_theta_transforms=True` (the default) is deprecated since Matplotlib 3.9. Support for this will be removed in Matplotlib in 3.11. To prevent this warning, set `apply_theta_transforms=False`, and make sure to shift theta values before being passed to this transform.
    tr = PolarAxes.PolarTransform()

tests/test_taylor.py::test_add_corr_grid
tests/test_taylor.py::test_add_corr_grid
  /Users/runner/miniconda3/envs/geocat_viz_build/lib/python3.12/site-packages/geocat/viz/taylor.py:423: MatplotlibDeprecationWarning: Passing `apply_theta_transforms=True` (the default) is deprecated since Matplotlib 3.9. Support for this will be removed in Matplotlib in 3.11. To prevent this warning, set `apply_theta_transforms=False`, and make sure to shift theta values before being passed to this transform.
    self.ax.vlines([np.arccos(value)],

tests/test_util.py::test_add_lat_lon_ticklabels
  /Users/runner/miniconda3/envs/geocat_viz_build/lib/python3.12/site-packages/cartopy/io/__init__.py:241: DownloadWarning: Downloading: https://naturalearth.s3.amazonaws.com/110m_physical/ne_110m_land.zip
    warnings.warn(f'Downloading: {url}', DownloadWarning)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/test_taylor.py::test_add_legend - Failed: Error: Image files did not match.
  RMS Value: 10.811921630003306
  Expected:  
    /var/folders/0j/bwqcs4y508s2n4ck4dhf3rpc0000gn/T/tmptwuhdz64/tests.test_taylor.test_add_legend/baseline.png
  Actual:    
    /var/folders/0j/bwqcs4y508s2n4ck4dhf3rpc0000gn/T/tmptwuhdz64/tests.test_taylor.test_add_legend/result.png
  Difference:
    /var/folders/0j/bwqcs4y508s2n4ck4dhf3rpc0000gn/T/tmptwuhdz64/tests.test_taylor.test_add_legend/result-failed-diff.png
  Tolerance: 
    2
================== 1 failed, [22](https://github.com/NCAR/geocat-viz/actions/runs/13710707456/job/38346433861#step:6:23) passed, 9 warnings in 23.32s ===================
```
</details>

--- a/docs/release-notes.rst
+++ b/docs/release-notes.rst
@@ -8,6 +8,10 @@ Release Notes
 v2024.xx.0 (unreleased)
 ---------------------------
 
+Bug Fixes
+^^^^^^^^^
+* Change Taylor diagram legend to be specified on the graphical axes to address legend location issues by `Katelyn FitzGerald`_ in (:pr:`273`)
+
 Documentation
 ^^^^^^^^^^^^^
 * Remove link to broken NCL_vector_5.py example by `Katelyn FitzGerald`_ in (:pr:`249`)

--- a/docs/release-notes.rst
+++ b/docs/release-notes.rst
@@ -10,7 +10,7 @@ v2024.xx.0 (unreleased)
 
 Bug Fixes
 ^^^^^^^^^
-* Change Taylor diagram legend to be specified on the graphical axes to address legend location issues by `Katelyn FitzGerald`_ in (:pr:`273`)
+* Change Taylor diagram legend to be specified on the graphical axes to address legend location issues by `Katelyn FitzGerald`_ in (:pr:`274`)
 
 Documentation
 ^^^^^^^^^^^^^

--- a/src/geocat/viz/taylor.py
+++ b/src/geocat/viz/taylor.py
@@ -719,12 +719,12 @@ class TaylorDiagram(object):
         if kwargs.get('labels') is None:
             labels = [p.get_label() for p in handles]
 
-        legend = self.ax.legend(handles,
-                                labels,
-                                loc=loc,
-                                bbox_to_anchor=(xloc, yloc),
-                                fontsize=fontsize,
-                                frameon=False)
+        legend = self._ax.legend(handles,
+                                 labels,
+                                 loc=loc,
+                                 bbox_to_anchor=(xloc, yloc),
+                                 fontsize=fontsize,
+                                 frameon=False)
         return legend
 
     def add_title(self,

